core/src/js/components/collection/hero-portraits.component.ts

```diff
@@ -218,57 +229,39 @@ export class HeroPortraitsComponent extends AbstractSubscriptionComponent implem
-						)}`;
-					case 2:
-						return `${storyIndex.padStart(2, '0')} - ${this.i18n.translateString(
-							'app.collection.filters.hero-portrait.book-of-mercs-chapters.xyrella',
-						)}`;
-					case 3:
-						return `${storyIndex.padStart(2, '0')} - ${this.i18n.translateString(
-							'app.collection.filters.hero-portrait.book-of-mercs-chapters.guff',
-						)}`;
-					case 4:
-						return `${storyIndex.padStart(2, '0')} - ${this.i18n.translateString(
-							'app.collection.filters.hero-portrait.book-of-mercs-chapters.kurtrus',
-						)}`;
-					case 5:
-						return `${storyIndex.padStart(2, '0')} - ${this.i18n.translateString(
-							'app.collection.filters.hero-portrait.book-of-mercs-chapters.tamsin',
-						)}`;
-					case 6:
-						return `${storyIndex.padStart(2, '0')} - ${this.i18n.translateString(
-							'app.collection.filters.hero-portrait.book-of-mercs-chapters.cariel',
-						)}`;
-					case 7:
-						return `${storyIndex.padStart(2, '0')} - ${this.i18n.translateString(
-							'app.collection.filters.hero-portrait.book-of-mercs-chapters.scabbs',
-						)}`;
-					case 8:
-						return `${storyIndex.padStart(2, '0')} - ${this.i18n.translateString(
-							'app.collection.filters.hero-portrait.book-of-mercs-chapters.tavish',
-						)}`;
-					case 9:
-						return `${storyIndex.padStart(2, '0')} - ${this.i18n.translateString(
-							'app.collection.filters.hero-portrait.book-of-mercs-chapters.brukan',
-						)}`;
-					case 10:
-						return `${storyIndex.padStart(2, '0')} - ${this.i18n.translateString(
-							'app.collection.filters.hero-portrait.book-of-mercs-chapters.dawngrasp',
-						)}`;
-					default:
-						return `${storyIndex.padStart(2, '0')} - ${this.i18n.translateString(
-							'app.collection.filters.hero-portrait.book-of-mercs-chapters.other',
-						)}`;
+				const BoMmatch = /BOM_(\d+)_.*/g.exec(refPortrait.id);
+				const BoMIndex = BoMmatch ? BoMmatch[1] : '';
+				if(+BoMIndex>=1 && +BoMIndex<=10){
+					return this.i18n.translateString(`app.collection.hero-portraits.book-of-mercs.${BoMIndex}`);
+				}
+				else{
+					return this.i18n.translateString(`app.collection.hero-portraits.book-of-mercs.other`);
+				}
+			case 'book-of-heroes':
+				const BoHmatch = /Story_(\d+)_.*/g.exec(refPortrait.id);
+				const BoHIndex = BoHmatch ? BoHmatch[1] : '';
+				if(+BoHIndex>=1 && +BoHIndex<=11){
+					return this.i18n.translateString(`app.collection.hero-portraits.book-of-heroes.${BoHIndex}`);
+				}
+				else{
+					return this.i18n.translateString(`app.collection.hero-portraits.book-of-heroes.other`);
 				}
 		}
 	}
 
+	private buildGroupIndex(category: CollectionPortraitCategoryFilter, refPortrait: ReferenceCard): number {
+		switch (category) {
+			case 'book-of-mercs':
+				const BoMmatch = /BOM_(\d+)_.*/g.exec(refPortrait.id);
+				return BoMmatch ? +BoMmatch[1] : 0;
+			case 'book-of-heroes':
+				const BoHmatch = /Story_(\d+)_.*/g.exec(refPortrait.id);
+				return BoHmatch ? +BoHmatch[1] : 0;
```

Comment:
Why return 0 instead of `null` when the index can't be found? This would also remove the need to hard-code the min/max indices (just check for null). Apart from that it looks good.